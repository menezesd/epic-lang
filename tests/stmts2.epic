func main() {
    x = 5;
    i = -1;
    for i in x..x*2 do print [41,i];
    print i; // must be -1 again
    y = 0;
    for i in x..x*2 do {
        print [42,i];
        print [i,43];
        y = y + 1;
    }
    print y;
    
    for i in 1..2 do
        print [44,i];
    for i in -8..0 do
        print [45,i];    
    for i in 10..10 do
        print [46,i];    
    for i in 10..1 do
        print [47,i];

    i0 = 15793317597137;
    while i0 != 0 do {
        print [48,i0,i0%2];
        i0 = i0/2;
    }
    while i0 != 42 do i0 = i0+1;
    print i0;
    while i0 != 44 do
    {
        i0 = i0+1;
    }
    print i0;

    for i in 0..10 do helper(i);
    for i in 0..10 do helper2(i);
}

func helper(x) {
    if x%3 == 0 then print [100,x];
    if x%2 == 0 then {
        print [[x]];
    } else {
        print [[[x]]];
    }
    if x%2 == 0 then print [[x]]; else print [[[x]]];
}

func helper2(x) {
    if x%3 == 0 then print[200,x];
    else if x%3 ==1 then print[201,x];
    else print[202,x];
    if x%3 == 0 then {
        print[200,x];
    } else if x%3 ==1 then {
        print[201,x];
    } else {
        print[202,x];
    }
}