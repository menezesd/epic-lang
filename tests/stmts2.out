[41, 5]
[41, 6]
[41, 7]
[41, 8]
[41, 9]
-1
[42, 5]
[5, 43]
[42, 6]
[6, 43]
[42, 7]
[7, 43]
[42, 8]
[8, 43]
[42, 9]
[9, 43]
5
[44, 1]
[45, -8]
[45, -7]
[45, -6]
[45, -5]
[45, -4]
[45, -3]
[45, -2]
[45, -1]
[48, 15793317597137, 1]
[48, 7896658798568, 0]
[48, 3948329399284, 0]
[48, 1974164699642, 0]
[48, 987082349821, 1]
[48, 493541174910, 0]
[48, 246770587455, 1]
[48, 123385293727, 1]
[48, 61692646863, 1]
[48, 30846323431, 1]
[48, 15423161715, 1]
[48, 7711580857, 1]
[48, 3855790428, 0]
[48, 1927895214, 0]
[48, 963947607, 1]
[48, 481973803, 1]
[48, 240986901, 1]
[48, 120493450, 0]
[48, 60246725, 1]
[48, 30123362, 0]
[48, 15061681, 1]
[48, 7530840, 0]
[48, 3765420, 0]
[48, 1882710, 0]
[48, 941355, 1]
[48, 470677, 1]
[48, 235338, 0]
[48, 117669, 1]
[48, 58834, 0]
[48, 29417, 1]
[48, 14708, 0]
[48, 7354, 0]
[48, 3677, 1]
[48, 1838, 0]
[48, 919, 1]
[48, 459, 1]
[48, 229, 1]
[48, 114, 0]
[48, 57, 1]
[48, 28, 0]
[48, 14, 0]
[48, 7, 1]
[48, 3, 1]
[48, 1, 1]
42
44
[100, 0]
[[0]]
[[0]]
[[[1]]]
[[[1]]]
[[2]]
[[2]]
[100, 3]
[[[3]]]
[[[3]]]
[[4]]
[[4]]
[[[5]]]
[[[5]]]
[100, 6]
[[6]]
[[6]]
[[[7]]]
[[[7]]]
[[8]]
[[8]]
[100, 9]
[[[9]]]
[[[9]]]
[200, 0]
[200, 0]
[201, 1]
[201, 1]
[202, 2]
[202, 2]
[200, 3]
[200, 3]
[201, 4]
[201, 4]
[202, 5]
[202, 5]
[200, 6]
[200, 6]
[201, 7]
[201, 7]
[202, 8]
[202, 8]
[200, 9]
[200, 9]
